pkg/util/limiter/query_limiter.go

```diff
@@ -65,18 +65,23 @@ func QueryLimiterFromContextWithFallback(ctx context.Context) *QueryLimiter {
 	return ql
 }
 
-// AddSeries adds the input series and returns an error if the limit is reached.
-func (ql *QueryLimiter) AddSeries(seriesLabels []cortexpb.LabelAdapter) error {
+// AddSeriesBatch adds the batch of input series and returns an error if the limit is reached.
+func (ql *QueryLimiter) AddSeries(series ...[]cortexpb.LabelAdapter) error {
```

Comment:
Just a small nit:

Can we receive an `[][]cortexpb.LabelAdapter` instaed of `...[]cortexpb.LabelAdapter` to avoid copying this slice over? 

From go doc:

>If f is [variadic](https://go.dev/ref/spec#Function_types) with a final parameter p of type ...T, then within f the type of p is equivalent to type []T. If f is invoked with no actual arguments for p, the value passed to p is nil. Otherwise, the value passed is a new slice of type []T with a new underlying array whose successive elements are the actual arguments, which all must be [assignable](https://go.dev/ref/spec#Assignability) to T. The length and capacity of the slice is therefore the number of arguments bound to p and may differ for each call site.